instrument/global.go

```diff
@@ -0,0 +1,35 @@
+package instrument
+
+import (
+	"sync"
+
+	"github.com/sirupsen/logrus"
+)
+
+var globalLock sync.Mutex
+var globalClient Client = MockClient{}
+
+func SetGlobalClient(client Client) {
+	if client == nil {
+		return
+	}
+	globalLock.Lock()
+	globalClient = client
+	globalLock.Unlock()
+}
+
+func GetGlobalClient() Client {
+	globalLock.Lock()
+	defer globalLock.Unlock()
+	return globalClient
+}
```

Comment:
so because `Client` is exposed, I think you're going to have some issues with pushing people to one or another. I think that it might be better to actually make the `Client` interface private, expose its methods directly in the package (like featureflag). Then you'd have a function like this:

```
func Identify(...blah...) {
  getGlobalClient().Indetify(blah)
}
```

It might be good to keep it exposed so people can know, but have package level functions make it slighlyt better ergonomics. For instance, logrus, metriks, and featureflag do something akin to this.

Reply:
cool will update 👍 (fyi featureflag is not doing that as I copy&pasted this from featureflag :P)

Reply:
actually nvm, I just found `func Enabled(key, userID string) bool` in featureflag package 🤦 apparently there is something I missed copy&pasta